src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -28,28 +29,32 @@ public class CriteriaQueryBuilder {
   private final TableNameGenerator tableNameGenerator;
   final TableVariable rootTable;
 
+  final SnapshotBuilderSettings snapshotBuilderSettings;
+
   private record OccurrenceTable(String tableName, String idColumnName) {}
 
-  private static final Map<String, OccurrenceTable> DOMAIN_TO_OCCURRENCE_TABLE =
+  private static final Map<Integer, OccurrenceTable> DOMAIN_TO_OCCURRENCE_TABLE =
       Map.of(
-          "Condition", new OccurrenceTable("condition_occurrence", "condition_concept_id"),
-          "Measurement", new OccurrenceTable("measurement", "measurement_concept_id"),
-          "Visit", new OccurrenceTable("visit_occurrence", "visit_concept_id"),
-          "Procedure", new OccurrenceTable("procedure_occurrence", "procedure_concept_id"),
-          "Observation", new OccurrenceTable("observation", "observation_concept_id"),
-          "Device", new OccurrenceTable("device_exposure", "device_concept_id"),
-          "Drug", new OccurrenceTable("drug_exposure", "drug_concept_id"));
-
-  private static OccurrenceTable getOccurrenceTableFromDomain(String domain) {
+          19, new OccurrenceTable("condition_occurrence", "condition_concept_id"),
```

Comment:
That sounds great! Is there a ticket for that work already?

Reply:
Oh, it looks like the only `SnapshotBuilderSettings` we have is the openapi model one. I think we'll need our own at some point (which will wrap the oepnapi one), but this can wait until then.